_posts/2019-01-25-building-apps-with-plug-router.md

```diff
@@ -0,0 +1,354 @@
+---
+author: Sean Callan
+categories: general
+date: 2019-01-25
+layout: post
+title: Building web apps with Plug.Router
+---
+
+When it comes to building a web application with Elixir many people will immediately reach for Phoenix.
+However, did you know `Plug.Router` is just as viable an option?
+Sometimes, it can be even faster.
+
+## The project
+
+For this project we'll build a simple single page portfolio site.
+We can expect our site to load and display our portfolio from a file, database, or somewhere else dynamically. As well as allowing users to submit contact information via a web form.
+
+__Please note__: To keep the application and tutorial concise, we will forego database backing and stub this out in favor of concentrating on the web portions.
+
+Want to skip the reading and just look at a code?
+Head over to [elixirschool/router_example](https://github.com/elixirschool/router_example).
+
+## Getting started
+
+To get started we need to do a couple of things:
+
+1. Generate a new project with `mix new --sup`
+1. Add the `plug_cowboy` dependency to `mix.exs`
+1. Put our router is into the supervision tree of our application
+
+Without further delay, let's get this show on the road and generate our new project:
+
+```shell
+$ mix new router_example --sup
+
+* creating README.md
+* creating .formatter.exs
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/router_example.ex
+* creating lib/router_example/application.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/router_example_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd router_example
+    mix test
+
+Run "mix help" for more commands.
+```
+
+Let's head into our new directory and open `mix.exs` in the editor of your choice.
+Here we'll make small a small change by adding the `plug_cowboy` dependency:
+
+```elixir
+defp deps do
+  [
+    {:plug_cowboy, "~> 2.0"}
+  ]
+end
+```
+
+With that change in place we can fetch our dependencies with `mix deps.get` and proceed.
+Though we have not yet created our router, let's get the supervisor setup out of the way.
+We'll need to open `lib/router_example/application.ex` next so we can update our supervisor's children.
+The `plug_cowboy` package makes this step easy with the included `Plug.Cowboy.child_spec/3` function.
+Let's update our application's `start/2` function:
+
+```elixir
+  def start(_type, _args) do
+    children = [
+      Plug.Cowboy.child_spec(scheme: :http, plug: RouterExample.Router, options: [port: 8080])
```

Comment:
The configured port `8080` does not match the port `4001` described at the bottom

Reply:
Doh! That's on me. I was finishing this up while doing some work-work and mixed my ports up.